client/src/utils/inputValidator.ts

```diff
@@ -1,30 +1,32 @@
-import type { InputValidation } from 'types';
+import type { InputValidation, InputValidator } from 'types';
 
 import { VALIDATOR, VALIDATION_ERRORS } from '../constants';
 
-const inputValidator = (inputValue: string, inputName: string, validation?: InputValidation) => {
-  const value = inputValue.trim();
-
-  let isValid = true;
-  let message = '';
-
+const inputValidator = (
+  inputValue: string,
+  inputName: string,
+  validation: Partial<InputValidation>
+): { valid: boolean; errMsg: string } => {
   if (validation) {
     const validators = Object.entries(validation);
 
-    isValid = !validators
-      .map(validator => {
-        const valid = VALIDATOR[validator[0]](value, validator[1]);
-
-        const err = valid ? '' : VALIDATION_ERRORS[validator[0]](inputName, validator[1]);
+    const result = validators.reduce(
+      ({ validatedList, errMsg }, [ruleName, ruleValue]) => {
+        const validated = VALIDATOR[ruleName as InputValidator](inputValue?.trim(), ruleValue);
+        const validationErr = VALIDATION_ERRORS[ruleName as InputValidator](inputName, ruleValue);
 
-        message += ` ${err}`;
+        return {
+          validatedList: [...validatedList, validated],
+          errMsg: `${errMsg} ${!validated ? validationErr : ''}`
+        };
+      },
+      { validatedList: [], errMsg: '' } as { validatedList: boolean[]; errMsg: string }
```

Comment:
please define types for `reduce` method correctly, here is the doc:
`reduce<U>(callbackfn: (previousValue: U, currentValue: T, currentIndex: number, array: T[]) => U, initialValue: U): U;`

Reply:
```
type ValidatorResult = { validatedList: boolean[]; errMsg: string[] };

const inputValidator = (
  inputValue: string,
  inputName: string,
  validation: Partial<InputValidation>
): { valid: boolean; errMsg: string } => {
  if (validation) {
    const validators = Object.entries(validation);

    const result = validators.reduce<ValidatorResult>(
      ({ validatedList, errMsg }, [ruleName, ruleValue]): ValidatorResult => {
        const validated = VALIDATOR[ruleName as InputValidator](inputValue?.trim(), ruleValue);
        const validationErr = VALIDATION_ERRORS[ruleName as InputValidator](inputName, ruleValue);

        return {
          validatedList: [...validatedList, validated],
          errMsg: [...errMsg, !validated ? validationErr : '']
        };
      },
      { validatedList: [true], errMsg: [''] }
    );

    return {
      valid: !result.validatedList.includes(false),
      errMsg: result.errMsg.join(' ')
    };
  }

  return { valid: true, errMsg: '' };
};
```
Is this better?

Reply:
much better :)
i have a question - do you really need to have empty string and true here instead of empty arrays?
 `{ validatedList: [true], errMsg: [''] }`

Reply:
I don't now, you're right:) Before I had to, it did not let me have it any other way.